### ISSUE TITLE
Compatibility issues with react-native-svg

### ISSUE THREAD

**Issue by matthieunelmes** [iyegoroff/react-native-color-matrix-image-filters#56]
RN: "0.64.2"
react-native-color-matrix-image-filters: "^5.2.10"

Amazing work here! I've been searching all over for a way to implement colormatrix filters in react-native.

However, react-native-svg <Image/> components do not render within <ColorMatrix/> components. To be honest, I suspect this is more due to how their Image components are structured:

`      <Image
                width="100"
                height="100"
                preserveAspectRatio="none"
                href={{ uri: vehicleDiagram.image }}
              />`
              
Compared to default RN Image components:

`      <Image
        style={styles.tinyLogo}
        source={require('@expo/snack-static/react-native-logo.png')}
      />`

**Comment by iyegoroff** [comment]
Thanks for the feedback!

Yes, you cannot wrap 'Image' component from `react-native-svg` because it has different structure on the native side.

**Comment by matthieunelmes** [comment]
Thank you for replying. I did attempt to start delving into the src to determine if this could be handled with a simple tweak. If you think it's possible, could you point me in the rough direction and I'm happy to attempt a PR. If you look at the react-native-svg repo, there is a huge demand for filters.

**Comment by iyegoroff** [comment]
To apply filtering this library searches for an image component inside filter component's children. https://github.com/iyegoroff/react-native-color-matrix-image-filters/blob/c62f3b229d90fbf481292f046f7dc6378556d758/ios/CMIFColorMatrixImageFilter.m#L48-L69 https://github.com/iyegoroff/react-native-color-matrix-image-filters/blob/c62f3b229d90fbf481292f046f7dc6378556d758/android/src/main/java/iyegoroff/RNColorMatrixImageFilters/ColorMatrixImageFilter.java#L37-L48 I guess it is possible to wrap a top 'Svg' component because it is backed by native image as far as I know. But this may not work out of the box.

**Comment by matthieunelmes** [comment]
No, I had to park this and move onto other things. SVG filters are a widely requested feature especially on react-native-svg.

**Comment by Wimmind** [comment]
> No, I had to park this and move onto other things. SVG filters are a widely requested feature especially on react-native-svg.

That is, at the moment there is no way to impose, for example, a black and white filter on the svg component from `react-native-svg` ?

**Comment by chunghn** [comment]
Hi. Did you find any workaround, either designing/coding aspects?

**Comment by matthieunelmes** [comment]
Not yet, I had to pause but I had a few ideas I wanted to test. Firstly was making use of the react-native-svg Use component to try importing a standard Image component with the filters applied.

**Comment by LoadJulz** [comment]
For anyone looking for a solution, first download the image and then display it using expo-image. This lets you wrap it in react-native-color-matrix-image-filters components. Why download it first? On iOS, expo-image doesn’t support SVGs loaded directly from a URL.

Here’s some example code:

`import React, { useState, useEffect } from 'react'
import { View, ActivityIndicator, Text } from 'react-native'
import { Image as ExpoImage } from 'expo-image'
import { Grayscale } from 'react-native-color-matrix-image-filters'

interface SvgViewerProps {
  width: string | number
  height: string | number
  uri: string
  style?: object
}

const SvgViewer: React.FC<SvgViewerProps> = ({ width, height, uri, style }) => {
  const [svgData, setSvgData] = useState<string | null>(null)
  const [isLoading, setIsLoading] = useState<boolean>(true)
  const [error, setError] = useState<string | null>(null)

  useEffect(() => {
    const fetchSvg = async () => {
      try {
        setIsLoading(true)
        const response = await fetch(uri)

        if (!response.ok) {
          throw new Error(`Failed to fetch SVG: ${response.status}`)
        }

        const svgText = await response.text()
        // Create a data URI from the SVG content
        const svgDataUri = `data:image/svg+xml;utf8,${encodeURIComponent(svgText)}`
        setSvgData(svgDataUri)
        setError(null)
      } catch (err) {
        console.error('Error fetching SVG:', err)
        setError(err.message)
      } finally {
        setIsLoading(false)
      }
    }

    fetchSvg()
  }, [uri])

  if (isLoading) {
    return null
  }

  if (error) {
    return <Text>Failed to load SVG: {error}</Text>
  }

  return (
    <Grayscale>
      <ExpoImage
        source={{ uri: svgData }}
        style={[{ width, height }, style]}
        contentFit="contain"
      />
    </Grayscale>
  )
}

export default SvgViewer
`
